src/Controller/MainController.java

```diff
@@ -113,42 +111,43 @@ public static void initialise() {
 								+ "and choose \"New File\".");
 					}
 				}
-
-			} else {
-				// TODO - fix this, as it is clearly a race condition
-				// This should never happen unless a file changes permissions
-				// or existence in the milliseconds that it runs the above code
-				// after checks in StartupController
-				UIManager.reportError("Failed to load file.");
+			} catch (FileNotFoundException e) {
```

Comment:
Please remove the "please contact support" from each of these.  I think that you misunderstood something that I said in class.  There are cases where it might be correct to put that into an error message presented to the user.  However, it looks like you have blindly added that to each catch block.  Think about what a user might do if they contacted support in the even of an IllegalBlockSizeException or an InvalidKeyException.

Before we start adding that to exception error messages, we would need a support forum or mailing list where we could direct users.  Right now adding "please contact support" is not giving the user anything actionable.